mdl/src/main/herd/scripts/installShepherd.sh

```diff
@@ -46,6 +46,7 @@ fi
 execute_cmd "echo \"From $0\""
 # Download and serve herd-ui (shepherd)
 execute_cmd "mkdir -p ${deployLocation}/herd-ui"
+#TODO failed sometimes, need retry on wget
```

Comment:
wget has implicit retries, did you want to add additional retry logic here?

Reply:
got error in this wget yesterday, I'm thinking to add additional logic if this happens very often. So add a TODO here, let me improve the TODO comments to mention this too.